### DISCUSSION TITLE
SOOP - Logging when using retry-max

### DISCUSSION THREAD

**Discussion by Infinitay** [streamlink/streamlink#6318] [General]
### Description

Would it be possible to cache login cookies when authenticating the SOOP plugin? That way there isn't a login request being made each time we attempt to fetch a stream. I noticed when looking at my log file that majority of the logs when a stream isn't live was filled with `Attempting to login using username and password` followed by the rest of the logs during the login process. It would be nice if instead of starting a new session each time we cache the retrieved cookies and only re-issue the login if the cookies will be expiring soon or some sort of auth error/403. Would this be possible?

**Comment by bastimeyer** [comment]
Going to close this, since the discussion forum is not a place for any requests.

If you look at the docs or plugin code, you can see that the plugin actually does cache logins:
- https://streamlink.github.io/plugins.html#soop
- https://github.com/streamlink/streamlink/blob/7.0.0/src/streamlink/plugins/soop.py#L280-L287

So if the authentication caching is broken, then please open a plugin issue on the issue tracker and include all the necessary details that are required, like a full debug log.

**Reply by Infinitay** [reply]
I'll create an issue in that case. Thank you for letting me know there was support for this.
